src/closure.jl

```diff
@@ -286,15 +288,30 @@ function enclose(exorig::Expr, reserve_per, minbatchsize, per::Symbol, mod)
     # end
   end
   closure = Symbol("##closure##")
+  threadlocal, threadlocal_type = threadlocal
+  threadlocal_init_single   = threadlocal == :() ? :() : :( $threadlocal_var = $threadlocal )
+  threadlocal_repack_single = threadlocal == :() ? :() : :( threadlocal = $threadlocal_type[threadlocal] )
+  threadlocal_init1         = threadlocal == :() ? :() : :( $threadlocal_var = Vector{$threadlocal_type}(undef, 0) )
+  threadlocal_init2         = threadlocal == :() ? :() : quote
+    resize!($(esc(threadlocal_var)),max(1,$(threadtup.args[2]))) # Why is the max necessary?
```

Comment:
This prepares the array of thread-local storage locations. Is there a better way to find how large to make the array?

Reply:
`max` is needed because `il` can be `0`.

This will be how many threads it actually ends up using, which can be less then the amount given by `threadtup.args`.
https://github.com/JuliaSIMD/Polyester.jl/blob/3def009e4250ab0831fbe0af9faf78ef443188ac/src/batch.jl#L216

Handling it in the way I suggested in an earlier comment could let you avoid creating the array until you're inside `batch`, after you know the number of threads.